backend/wmg/data/cube_pipeline.py

```diff
@@ -75,13 +81,19 @@ def load_data_and_create_cube(
     logger.info("Loaded datasets into corpus")
     create_cubes(corpus_path)
     logger.info("Built expression summary cube")
-
+    if validate_cubes:
+        is_valid = Validation(corpus_path).validate_cube()
+        if is_valid is False:
+            return
     cell_type_by_tissue = get_cell_types_by_tissue(corpus_path)
     generate_cell_ordering(snapshot_path, cell_type_by_tissue)
     generate_primary_filter_dimensions(snapshot_path, corpus_name, timestamp)
     logger.info("Generated cell ordering json file")
-    update_s3_resources(snapshot_path, timestamp)
+    upload_artifacts_to_s3(snapshot_path, timestamp)
     logger.info("Copied snapshot to s3")
+    if validate_cubes:
+        update_s3_resources(timestamp)
+    logger.info(f"Updated latest_snapshot_identifier in s3. Current snapshot id is {timestamp}")
```

Comment:
⚠️ Indent this line or move it to `update_s3_resources`, if keeping `if` stmt.

Reply:
This will copy the resources to s3 to make them available for debugging but it will skip updating the snapshot if the cube wasnt validated

Reply:
ahh good catch. Thanks!